### PR TITLE
No. Series: Ability to show only expiring lines

### PR DESCRIPTION
**Pull request by AndreasMoth**
This PR introduces the ability to see No. Series that are expiring. Whether that means there are no open lines or at least one line is in warning state.

Fixes [AB#476364](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/476364)

### CODE REVIEW

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 1]
```diff
@@ -241,6 +241,37 @@ page 456 "No. Series"
...
+            {
+                Caption = 'View';
+                ShowAs = SplitButton;
+
+                action(ShowAll)
+                {
+                    ApplicationArea = Basic, Suite;
+                    Caption = 'Show All';
+                    Image = List;
+                    ToolTip = 'Show all No. Series.';
```
Comment (NKarolak) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: Aligning with the other tooltips: no abbreviations.
```suggestion
                    ToolTip = 'Show all number series.';
```

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 2]
```diff
@@ -241,6 +241,37 @@ page 456 "No. Series"
...
+                        ShowNoSeriesWithWarningsOnly(false);
+                    end;
+                }
+
+                action(ShowExpiring)
+                {
+                    ApplicationArea = Basic, Suite;
+                    Caption = 'Show Expiring';
+                    Image = ShowList;
+                    ToolTip = 'Show No. Series that have no lines or an open line which have reached the warning threshold.';
```
Comment (NKarolak) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]:
```suggestion
                    ToolTip = 'Show number series that have no lines or an open line which have reached the warning threshold.';
```

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 3]
```diff
@@ -319,4 +351,38 @@ page 456 "No. Series"
         NoSeriesSingle := Implementation;
         AllowGaps := NoSeriesSingle.MayProduceGaps();
     end;
+
+    local procedure ShowNoSeriesWithWarningsOnly(ShowWarningsOnly: Boolean)
```
Comment (grobyns) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: You should move this to No. Series - Setup Impl. codeunit

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 4]
```diff
@@ -292,4 +324,38 @@ page 456 "No. Series"
...
+    var
+        NoSeriesLine: Record "No. Series Line";
+        NoSeries: Codeunit "No. Series";
+        LastNoUsedForLine: Code[20];
+    begin
+        ShowNoSeriesWithWarnings := ShowWarningsOnly;
```
Comment (NKarolak) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: What do we need the global variable `ShowNoSeriesWithWarnings` for?

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 5]
```diff
@@ -319,4 +351,38 @@ page 456 "No. Series"
...
+        Rec.ClearMarks();
+        if not ShowNoSeriesWithWarnings then begin
+            Rec.MarkedOnly(false);
```
Comment (grobyns) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: I think you should set MarkedOnly(false) always, so above the if

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al (hunk 4 above)
Comment (AndreasMoth) [reply src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]: Was needed in a previous design but not anymore, will remove

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al [hunk 6]
```diff
@@ -259,6 +259,37 @@ page 456 "No. Series"
...
+                        ShowNoSeriesWithWarningsOnly(false);
+                    end;
+                }
+
+                action(ShowExpiring)
+                {
+                    ApplicationArea = Basic, Suite;
+                    Caption = 'Show Expiring';
+                    Image = ShowList;
+                    ToolTip = 'Show number series that have no lines or an open line which have reached the warning threshold.';
```
Comment (ikoletic) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeries.Page.al]:
```suggestion
                    ToolTip = 'Show number series that require your attention, such as, number series that have no lines or an have lines which have reached the warning threshold.';
```

**Review by ikoletic — CHANGES_REQUESTED** [review]

**Review by grobyns — CHANGES_REQUESTED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Review by darjoo — APPROVED** [review]

File: src/Business Foundation/App/NoSeries/src/Setup/NoSeriesSetupImpl.Codeunit.al [hunk 7]
```diff
@@ -63,6 +63,36 @@ codeunit 305 "No. Series - Setup Impl."
         Implementation := NoSeriesLine.Implementation;
     end;
 
+    procedure ShowNoSeriesWithWarningsOnly(var NoSeries: Record "No. Series"; ShowWarningsOnly: Boolean)
```
Comment (darjoo) [inline src/Business Foundation/App/NoSeries/src/Setup/NoSeriesSetupImpl.Codeunit.al]: Would be better to have two functions
`ShowNoSeries` and `ShowNoSeriesWithWarningsOnly`. Having a boolean parameter to determine whether it should show warnings only defeats the purpose of the function name as I expect it to always show with warnings only.

**Review by darjoo — APPROVED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Pull request merged**